control-plane/csi-driver/src/context.rs

```diff
@@ -62,6 +66,26 @@ impl Parameters {
             None => None,
         })
     }
+    fn parse_map(
+        value: Option<&String>,
+    ) -> Result<Option<HashMap<String, String>>, serde_json::Error> {
+        Ok(match value {
+            Some(json_str) => {
+                let value: Value = serde_json::from_str(json_str)?;
+                // Convert Value to Map
+                if let Value::Object(map) = value {
```

Comment:
```suggestion
                if let Some(obj) = value.as_object() {
```

Reply:
We should handle v.as_str().unwrap() better, should return an error rather than crash...